### PR TITLE
use adr-tools and a custom template for our ADR

### PR DESCRIPTION
**Pull request by misaka**
### Context

Wherever we decide to put ADRs, a little bit of tooling can help us keep things consistent.

### Changes proposed in this pull request

Add npryce's adr-tools to help manage ADRs.

### CODE REVIEW

File: docs/adr/0002-use-adr-tools-with-custom-template.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+- simple, well understood
+- bash scripts
+- flexible enough for us (can customise homedir, templates)
+- also being used by Apply
+
+#### Cons
+
+- unknown
+
+### 2. Markdown Architectural Decision Records (mard)
```
Comment (defong) [inline docs/adr/0002-use-adr-tools-with-custom-template.md]: Mard? Typo

Comment (misaka) [reply docs/adr/0002-use-adr-tools-with-custom-template.md]: Only about the 1000th time I did that.

File: docs/adr/0002-use-adr-tools-with-custom-template.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#### Cons
+
+- unknown
+
+### 2. Markdown Architectural Decision Records (madr)
+
+These are a branch of npryce's adr-tools. See
+[adr/madr](https://github.com/adr/madr). They seem to have added a whole bunch
+of extra stuff that I'm not sure we need, although I do like some of the ideas
+in their ADR template which I want to nick for our own template.
```
Comment (defong) [inline docs/adr/0002-use-adr-tools-with-custom-template.md]: Maybe utilise rather then nick

Comment (dankmitchell) [reply docs/adr/0002-use-adr-tools-with-custom-template.md]: Never thought id see the day when @defong would be correcting someone else on their language 😄

Comment (defong) [reply docs/adr/0002-use-adr-tools-with-custom-template.md]: for posterity I did kinda approve of @misaka nicking something before correcting his behaviour...

**Pull request merged**